tensorly/metrics/entropy.py

```diff
@@ -18,6 +18,8 @@ def vonneumann_entropy(tensor):
     -------
     von_neumann_entropy : order-0 tensor
     """
+    square_dim = int(tl.sqrt(tl.prod(tl.tensor(tensor.shape))))
```

Comment:
It may be worth documenting this in the docstring and specifying the possible sizes of tensor (either directly a matrix or a tensor satisfying prod(t.shape) = N**2).